src/server/service/debug.rs

```diff
@@ -0,0 +1,208 @@
+use storage::{CF_DEFAULT, CF_LOCK, CF_RAFT, CF_WRITE};
+
+pub type Callback = Box<FnBox(Result<Response, Error>) + Send>;
+
+#[derive(Clone)]
+pub struct Service {
+    scheduler: Scheduler<Request>,
+}
+
+impl Service {
+    pub fn new(scheduler: Scheduler<Request>) -> Service {
+        Service { scheduler }
+    }
+}
+
+fn make_callback<T: Debug + Send + 'static>() -> (Box<FnBox(T) + Send>, oneshot::Receiver<T>) {
+    let (tx, rx) = oneshot::channel();
+    let callback = move |resp| { tx.send(resp).unwrap(); };
+    (box callback, rx)
+}
+
+impl debugpb_grpc::Debug for Service {
+    fn get(&self, ctx: RpcContext, mut req: GetRequest, sink: UnarySink<GetResponse>) {
+        const LABEL: &'static str = "debug_get";
+
+        let (cb, future) = make_callback();
+        let req = Request::Get {
+            cf: req.get_cf(),
+            key_encoded: req.take_key_encoded(),
+            callback: cb,
+        };
+        let on_error = |e| {
+            debug!("{} failed: {:?}", LABEL, e);
+        };
+        if self.scheduler.schedule(req).is_ok() {
+            let future = future.then(|v| match v {
+                Ok(Ok(Response::Get { value })) => {
+                    let mut resp = GetResponse::new();
+                    resp.set_value(value);
+                    sink.success(resp).map_err(on_error)
+                }
+                Ok(Err(Error::NotFound(msg))) => {
+                    let status = RpcStatus::new(RpcStatusCode::NotFound, Some(msg));
+                    sink.fail(status).map_err(on_error)
+                }
+                Ok(Err(Error::InvalidArgument(msg))) => {
+                    let status = RpcStatus::new(RpcStatusCode::InvalidArgument, Some(msg));
+                    sink.fail(status).map_err(on_error)
+                }
+                Err(canceled) => {
+                    let status =
+                        RpcStatus::new(RpcStatusCode::Unknown, Some(format!("{:?}", canceled)));
+                    sink.fail(status).map_err(on_error)
+                }
+                _ => {
+                    unreachable!();
+                }
+            });
+            ctx.spawn(future);
+        } else {
+            let status = RpcStatus::new(RpcStatusCode::Unavailable, None);
+            ctx.spawn(sink.fail(status).map_err(on_error));
+        }
+    }
+
+    fn mvcc(&self, _: RpcContext, _: MvccRequest, _: UnarySink<MvccResponse>) {
+        unimplemented!()
+    }
+
+    fn raft_log(&self, _: RpcContext, _: RaftLogRequest, _: UnarySink<RaftLogResponse>) {
+        unimplemented!()
+    }
+
+    fn region_info(&self, _: RpcContext, _: RegionInfoRequest, _: UnarySink<RegionInfoResponse>) {
+        unimplemented!()
+    }
+
+    fn size(&self, _: RpcContext, _: SizeRequest, _: UnarySink<SizeResponse>) {
+        unimplemented!()
+    }
+
+    fn scan(&self, _: RpcContext, _: ScanRequest, _: ServerStreamingSink<ScanResponse>) {
+        unimplemented!()
+    }
+}
+
+pub struct Runner {
```

Comment:
Can we let `tikv-ctl` use `Runner` directly for localstore? Do you think it's a good idea?

Reply:
For now I thinks `tikv-ctl` can't use `Runner` directly, because `on_get` returns nothing. So I thinks we should seprate `Runner` and `MessageProcessor`(or something with other name).

Reply:
Not always the same, but I prefer using the same code to fetch RocksDB data, not through coprocessor or scheduler in the gRPC API.

But Runner should also supply some APIs which `tikv-ctl` can't use, like showing metrics or showing current store status, etc.